app/models/course.rb

```diff
@@ -304,4 +305,11 @@ def ensure_required_params
     self.timeline_start ||= start
     self.timeline_end ||= self.end
   end
+
+  def set_end_times
+    unless self.use_start_and_end_times
+      self.end = self.end.end_of_day
+      self.timeline_end = self.timeline_end.end_of_day
```

Comment:
No need for self on the right side of the assignment here. (Still needed for `self.end` because end is a special ruby word.)